### PR TITLE
[Clang] [NFC] Add release note about libstdc++ bug

### LINKED ISSUE #92439: [Clang] `++this` is accepted when it’s not instantiated
This code compiles, unless we instantiate and call `f()`. This doesn’t seem right to me because, irrespective of what the type of e.g. `this` ends up being, it’s a prvalue, and we can’t increment that.

```c++
template <typename T>
struct S : T {
    auto t() -> S*;
    void f() { 
        ++reinterpret_cast<S*>(4);
        ++this;
        ++t(); 
    }
};
```

Imo we should diagnose this earlier if that is feasible. GCC also suffers from the same problem, which caused `++this` as a typo for `++*this` to go unnoticed in a libstdc++ header because the template containing it was never instantiated by libstdc++. MSVC also accepts this code. Is this a bug or a standards defect? Because I don’t see a situation where `++this` would be semantically valid.

CC @AaronBallman, @cor3ntin, @erichkeane

### PR DESCRIPTION
**Pull request by Sirraide**
Adding a release note about this as discussed in #92439. 

(This is an NFC change, but I’m adding you as reviewers in case there is anything you want to add to this since you were involved in the discussion.)

### CODE REVIEW

File: clang/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -59,6 +59,18 @@ C++ Specific Potentially Breaking Changes
...
+  on systems using that version of libstdc++ and Clang 19, with an error that looks
+  something like this:
+
+  .. code-block:: text
+
+    <source>:4:5: error: expression is not assignable
+    4 |     ++this;
+      |     ^ ~~~~
+
+  To fix this, update libstdc++ to a newer version, such as 14.1.1 or 14.2.
```
Comment (cor3ntin) [inline clang/docs/ReleaseNotes.rst:72]:
```suggestion
  To fix this, update libstdc++ to version 14.1.1 or greater.
```
14.2 _should_ release before clang 19 but probably not by much so I don't think we want to be that specific

**Comment by cor3ntin** [comment]
I would keep that issue and just remove the fixes (We can still link the issue to this PR by posting a link in a comment)

**Comment by Sirraide** [comment]
> I would keep that issue and just remove the fixes 

Done

**Comment by cor3ntin** [comment]
(I also think that the libstdc++ bug demonstrated some usefulness for a `++this` -> `++*this` fixit... maybe?)

**Review by cor3ntin — APPROVED** [review]
Thanks for doing this!

**Comment by Sirraide** [comment]
> (I also think that the libstdc++ bug demonstrated some usefulness for a `++this` -> `++*this` fixit... maybe?)

That does sound like it might be useful; I’m going to open another issue about that so we don’t forget about it.

**Comment by Sirraide** [comment]
I’ve opened an issue to add a fix-it hint and tagged it as a good first issue because for once this is something that seems like it would be fairly straight-forward to do: #93066

**Pull request merged**